code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/SmallImageCard.kt

```diff
@@ -33,16 +47,19 @@ fun SmallImageCard(
     style: SmallImageUIStyle,
     observer: AepUIEventObserver?,
 ) {
+    var bitmap by remember { mutableStateOf<Bitmap?>(null) }
+    val imageUrl = if (isSystemInDarkTheme() && ui.getTemplate().image?.darkUrl != null)
+        ui.getTemplate().image?.darkUrl else ui.getTemplate().image?.url
 
-    // TODO - Add id for LaunchedEffect, test if it is working
-    LaunchedEffect(key1 = Unit) {
+    LaunchedEffect(ui.getTemplate().id) {
         observer?.onEvent(UIEvent.Display(ui))
     }
 
-    // TODO - Add id for DisposableEffect, test if it is working
-    DisposableEffect(key1 = Unit) {
-        onDispose {
-            observer?.onEvent(UIEvent.Dismiss(ui))
+    LaunchedEffect(imageUrl) {
```

Comment:
I think we can combine the two LaunchedEffect statements into one based on the id, given that the url and id do not change across recompositions

Reply:
The imageUrl will change based on dark and light mode, so used a different launched effect

Reply:
Actually dark mode change triggers entire view recomposition, will change to use a single launched effect 👍